app/src/main/java/org/simple/clinic/registration/pin/RegistrationPinScreenController.kt

```diff
@@ -55,11 +66,12 @@ class RegistrationPinScreenController @Inject constructor(
   }
 
   private fun updateOngoingEntryAndProceed(events: Observable<UiEvent>): Observable<UiChange> {
-    val pinTextChanges = events.ofType<RegistrationPinTextChanged>()
     val doneClicks = events.ofType<RegistrationPinDoneClicked>()
+    val pinTextChanges = events
+        .ofType<RegistrationPinTextChanged>()
+        .map { it.pin }
 
-    return doneClicks
-        .withLatestFrom(pinTextChanges.map { it.pin })
+    return Observables.combineLatest(doneClicks, pinTextChanges)
```

Comment:
Why was this done here? Won't this let us skip the need to filter on a valid PIN since `RegistrationPinDoneClicked` will emit whenever a valid PIN is entered?

```kotlin
return doneClicks
.wlf(pinTextChanges) { _, pin -> pin }
// Do whatever
```

Reply:
Nvm, I just realised that the event can be triggered from the IME Done action too.

Reply:
Because `Done` is now a derived event that is automatically emitted when `Text` length reaches 4. As a result, `doneClicks.withLatestFrom(textChanges)` no longer works. `textChanges.withLatestFrom(doneClicks)` might have worked.